### PR TITLE
remove some lower bounds in ArrayOps

### PR DESCRIPTION
**Pull request by xuwei-k**
We can remove lower bounds in ArrayOps because ArrayOps is invariant and not collection.

ref https://github.com/scala/bug/issues/10831

### CODE REVIEW

File: src/repl-frontend/scala/tools/nsc/interpreter/shell/Reporter.scala [hunk 1]
```diff
@@ -150,7 +150,7 @@ class ReplReporterImpl(val config: ShellConfig, val settings: Settings = new Set
   // TODO: the console could be empty due to external changes (also, :reset? -- see unfortunate example in jvm/interpeter (plusOne))
   def printMessage(posIn: Position, msg: String): Unit = {
     if ((posIn eq null) || (posIn.source eq NoSourceFile)) printMessage(msg)
-    else if (posIn.source.file.name == "<console>" && posIn.line == 1 && posIn.source.content.indexOf("\n") == -1) {
+    else if (posIn.source.file.name == "<console>" && posIn.line == 1) {
```
Comment (julienrf) [inline src/repl-frontend/scala/tools/nsc/interpreter/shell/Reporter.scala:153]: Why this change?

Comment (xuwei-k) [reply src/repl-frontend/scala/tools/nsc/interpreter/shell/Reporter.scala:153]: `posIn.source.content` is `Array[Char]`.  `posIn.source.content.indexOf("\n")` always return `-1`. 

https://scala-ci.typesafe.com/job/scala-2.13.x-validate-main/3457/console

```
[error] /home/jenkins/workspace/scala-2.13.x-validate-main/src/repl-frontend/scala/tools/nsc/interpreter/shell/Reporter.scala:153: type mismatch;
[error]  found   : String("\n")
[error]  required: Char
[error]     else if (posIn.source.file.name == "<console>" && posIn.line == 1 && posIn.source.content.indexOf("\n") == -1) {
[error]                                                                                                       ^
```

Comment (julienrf) [reply src/repl-frontend/scala/tools/nsc/interpreter/shell/Reporter.scala:153]: Should we write `indexOf('\n')` instead?

Comment (szeiger) [reply src/repl-frontend/scala/tools/nsc/interpreter/shell/Reporter.scala:153]: I recommend adding a test for this case. Checking for `indexOf('\n')` was probably the intention behind this code but with the current implementation it wouldn't do anything.

Comment (xuwei-k) [reply src/repl-frontend/scala/tools/nsc/interpreter/shell/Reporter.scala:153]: some tests failed if change to `indexOf('\n')` 🤔 

https://scala-ci.typesafe.com/job/scala-2.13.x-validate-main/3459/#showFailuresLink

screen shot 2018-08-04 at 15 54 37

**Comment by xuwei-k** [comment]
Anyway I have revert `indexOf` and `src/repl-frontend/scala/tools/nsc/interpreter/shell/Reporter.scala` changes. PTAL :pray:

**Comment by szeiger** [comment]
Well, judging by the diff from one of the tests *not* checking for `'\n'` is the correct thing to do. You can clearly see the difference in line handling:

```
% diff /Users/szeiger/code/scala/test/files/run/t1931.check /Users/szeiger/code/scala/test/files/run/t1931-run.log
@@ -10,8 +10,9 @@ scala> import Predef.{ any2stringadd => _, _ }
 import Predef.{any2stringadd=>_, _}

 scala> x + " works"
+       x + " works"
          ^
-       error: value + is not a member of Any
+On line 1: error: value + is not a member of Any

 scala> import Predef._
 import Predef._
@@ -27,8 +28,9 @@ scala> import Predef._
 import Predef._

 scala> f
+       f
        ^
-       error: not found: value f
+On line 1: error: not found: value f

 scala> Predef.f
 res4: Int = 42
```

I'd say the code in `Reporter` was "correct" and simply removing the extraneous check (like you originally did) was the right way to go. We should remove the type param on `indexOf` because it caught this dead code.

**Review by szeiger — APPROVED** [review]
